test: lib: Skip tests/lib/cmsis_dsp/matrix on MEC17.

Kunli123 opened the pull request:
The SRAM for date segment in MEC17 is only 64KB.
    malloc allocated 12.8KB twice,
    calloc allocated 24.9KB.
    Update the test to ZTEST new API also consumes more memory,
    and the memory allocation failed at the last malloc.
    So skip this test on MEC17.

Fixes #50452

Signed-off-by: Kun Li <kun1x.li@intel.com>

stephanosio reviewed (changes requested):
> The calloc function requires large memory,Allocating after is easy to fail.

Why? The rationale given here does not make much sense to me.

Sure, this "works" because of how the newlib malloc algorithm is implemented and it just happens to request a smaller chunk later through `sbrk` if `calloc(length, ...` is called first in this particular case, but there is nothing that guarantees this behaviour and is subject to change with a different version/configuration of newlib and/or with a different libc.

The fact that this test may fail depending on the order of calloc/malloc call implies that there is very little room left on this device with 64KB RAM running this test, and this test may very well fail again on this platform with a slight change in the RAM usage in the future due to other changes.

For that reason, I do not think this is the right way to go about resolving this issue. My suggestion would be:

a) look into a way to actually reduce RAM usage, OR
b) increase `min_ram: 64` to a larger value (e.g. `min_ram: 72`) so that this test is filtered out for `mec172xevb_assy6906` and other platforms with 64KB RAM, which has been demonstrated to be not so sufficient for this test.

Obviously, "a" would require more extensive changes to the internals of this test, so I would highly recommend implementing "b" as I initially suggested in https://github.com/zephyrproject-rtos/zephyr/issues/50452#issuecomment-1259748729.

Kunli123 committed:
tests: lib: Skip tests/lib/cmsis_dsp/matrix on MEC17

The SRAM for date segment in MEC17 is only 64KB.
malloc allocated 12.8KB twice,
calloc allocated 24.9KB.
Update the test to ZTEST new API also consumes more memory,
and the memory allocation failed at the last malloc.
So increase the memory to 128kB. The boards with physical
memory less than 128kB will skip the test automatically.

Fixes #50452
Signed-off-by: Kun Li <kun1x.li@intel.com>

stephanosio commented:
Closing this since https://github.com/zephyrproject-rtos/zephyr/pull/51883 has already addressed this issue.
